### DISCUSSION TITLE
[Idea] Merge autoupgrade module into PrestaShop core

### DISCUSSION THREAD

**Discussion by cruftex** [PrestaShop/PrestaShop#31288] [Ideas]
### The problem

I am a new PrestaShop user and recently did an upgrade from 8.0.0 to 8.0.1. My naive thinking was, that I don't need the auto upgrade module, and I can do a manual upgrade instead. However to my surprise all relevant code for upgrades is actually in the autoupgrade module and not included in PrestaShop.

To my confusion the module is also marketed as 1 click upgrade and referred to as upgrade assistant in the dev docs.

To my further surprise the autoupdate module is also needed when doing a manual update.

The autoupgrade module has a separate issue tracker, however, there are also a lot of issues regarding the autoupgrade in in the `PrestaShop` project.

### The solution

What about merging the autoupgrade code into the core?

### Alternatives

Smaller steps could minimize confusion first by:
- Consistent naming of the module
- Close issue tracker of autoupgrade module and refer to main issue tracker

### Do you plan to work on this subject?

- [ ] I'm willing to contribute a [formal specification](https://github.com/PrestaShop/prestashop-specs).
- [ ] I'm willing to provide any wireframes or design assets required for this feature.
- [ ] I'm willing to submit a Pull Request that implements this feature.
- [X] I'm willing to help verify that the implemented feature works as intended and produces no unintended side effects.

**Comment by matks** [comment]
Hello there @cruftex I understand the idea but there are multiple reasons why the upgrade system is inside a module

1. It allows people who dont want it to simply uninstall it. We dont want un-used code to stay on your shop
2. It allows to release bug fixes and improvements of the upgrade system without having to do a new release of the Core. New releases of the Core are very expensive and long while a module release can be a log easier and lighter (then quicker).
3. Core releases are frozen in time: the code of PrestaShop 1.7.7.1 will never change it is a rule of open source. So if we want to fix something related to upgrade from 1.7.7.1 to 8.0.0 we have to do it somewhere outside the Core. The upgrade module allows us to do that.

So the fact it is a module is actually on purpose mostly to allow the upgrade module to evolve in an easier and smoother way 😉

**Comment by cruftex** [comment]
Hi @matks. Thanks for the details and background information! I agree why you don't want to move it to core. Maybe there is still little things that can be done, to avoid confusion. Ideas:
- Consolidate module naming At the moment its named upgrade assistant, 1-click upgrade and auto upgrade
- Improve documentation, that the autoupgrade module is always needed for the upgrade, that is tracked here: https://github.com/PrestaShop/docs/issues/657
- And maybe close issue tracker of the autoupgrade GitHub project?
